src/components/Checkout.jsx

```diff
@@ -0,0 +1,80 @@
+import { useState } from "react";
+
+export default function Checkout({ setCartArr }) {
+  const inputs = {
+    firstName: "",
+    lastName: "",
+    email: "",
+    zipCode: "",
+  };
+  const [inputState, setInputState] = useState(inputs);
+  function renderTextChange(e, inputKey) {
+    const inputClone = { ...inputState };
+    inputClone[inputKey] = e.target.value;
+    setInputState(inputClone);
+  }
+
+  function checkingOut(e) {
+    e.preventDefault();
+    console.log(e);
+    const nameValidation = /^[a-zA-Z\s]+$/;
+    const emailValidation = /\S+@\S+\.\S+/;
+    if (!nameValidation.test(e.target[0].value)) {
```

Comment:
Because you have `if` statements and not `else if` statements, the user night get one or all alerts but still be able to submit the form and checkout.